src/api/uploads/__tests__/MultiputPart-test.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-unused-expressions, no-underscore-dangle */
+import MultiputPart, { PART_STATE_UPLOADED } from '../MultiputPart';
+
+const sandbox = sinon.sandbox.create();
+
+describe('api/MultiputPart', () => {
+    const options = {};
+    const index = 0;
+    const offset = 0;
+    const size = 1;
+    const sessionId = 1;
+    const sessionEndpoints = {};
+    const config = {};
+    const getPartsState = sandbox.stub();
+    const getNumPartsUploading = sandbox.stub();
+    let MultiputPartTest;
+    beforeEach(() => {
+        MultiputPartTest = new MultiputPart(
+            options,
+            index,
+            offset,
+            size,
+            sessionId,
+            sessionEndpoints,
+            config,
+            getPartsState,
+            getNumPartsUploading
+        );
+    });
+
+    afterEach(() => {
+        sandbox.verifyAndRestore();
+    });
+
+    describe('upload()', () => {
+        it('should noop if sha256 is not available', () => {
```

Comment:
Trying to upload the part before the SHA-256 is ready would imply a logic error in the code, which is really bad.  In this case I think it makes sense to fail more loudly (with an exception) as the expectation is that it should never happen, and we definitely want to know if it does.

Reply:
done